fem/libceed/diffusion.cpp

```diff
@@ -23,7 +23,12 @@ void CeedPADiffusionAssemble(const FiniteElementSpace &fes,
                              const mfem::IntegrationRule &irm, CeedData& ceedData)
 {
 #ifdef MFEM_USE_CEED
-   CeedInt dim = fes.GetMesh()->SpaceDimension();
+   Mesh &mesh = *fes.GetMesh();
+   // Perform checks for some assumptions made in the Q-functions.
+   MFEM_VERIFY(mesh.Dimension() == mesh.SpaceDimension(), "case not supported");
```

Comment:
@v-dobrev If you know the formula for the other cases, I can implement them in an other PR.

Reply:
Same here, no need to do it right now.

Reply:
For completeness: if `J` is the `SpaceDimension` x `Dimension` Jacobian matrix, then gradients are mapped from reference to physical space using the matrix `J (J^T J)^{-1}`, so the expression for a quadrature point becomes
```
[det(J^T J)]^{1/2} (J^T J)^{-1} J^T Q J (J^T J)^{-1}
```
where `Q` is the diffusivity coefficient matrix. Substituting `(J^T J)^{-1} = adj(J^T J)/det(J^T J)`, the expression can be simplified as
```
[det(J^T J)]^{-3/2} X^T Q X,   where X = J adj(J^T J)
```
When `J` is square, `J (J^T J)^{-1} = J^{-T} = adj(J)^T/det(J)` and the simplified form is
```
[det(J)]^{-1} Y^T Q Y,   where Y = adj(J)^T
```